api/src/main/java/org/openmrs/api/impl/ObsServiceImpl.java

```diff
@@ -109,8 +105,14 @@ public Obs saveObs(Obs obs, String changeMessage) throws APIException {
 			setPersonFromEncounter(obs);
 			return saveObsNotDirty(obs, changeMessage);
 		} else {
-			setPersonFromEncounter(obs);
-			return saveExistingObs(obs,changeMessage);
+
+			if(Objects.equals(obs.getPersonId(), obs.getEncounter().getId())){
```

Comment:
The code should be in ObsValidator and you should be comparing person objects and not their personIds

Reply:
Ok I will do the necessary changes.  thanks

Reply:
@wluyima I wrote the code to check for person objects in ObsValidator and placed the test in ObsValidatorTest .Should I call the ObsValidator's method in ObsServiceImpl ?

Reply:
Why? Move all the code to the validator, there should no changed in ObsServiceImpl